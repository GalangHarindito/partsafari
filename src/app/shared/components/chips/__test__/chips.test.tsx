import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Chips from '../chips';
import '@testing-library/jest-dom';

describe('Chips Component', () => {
  const mockOnClick = jest.fn();

  test('renders with the correct title', () => {
    const { container } = render(
      <Chips title="Test Title" active="Test Title" onClick={mockOnClick} />
    );

    const chipElement = screen.getByText(/TEST TITLE/i);
    expect(chipElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('applies active class', () => {
    const { container } = render(
      <Chips title="Active Chip" active="Active Chip" onClick={mockOnClick} />
    );

    expect(container.firstChild).toHaveClass('bg-tab-active');
  });

  test('applies inactive class', () => {
    const { container } = render(
      <Chips title="Inactive Chip" active="Active Chip" onClick={mockOnClick} />
    );

    expect(container.firstChild).toHaveClass('bg-tab-inactive');
  });

  test('calls onClick function', () => {
    render(
      <Chips
        title="Clickable Chip"
        active="Clickable Chip"
        onClick={mockOnClick}
      />
    );

    const chipElement = screen.getByText(/CLICKABLE CHIP/i);
    fireEvent.click(chipElement);

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
