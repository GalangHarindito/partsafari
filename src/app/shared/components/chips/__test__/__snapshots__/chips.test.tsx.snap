// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chips Component renders with the correct title 1`] = `
<div>
  <div
    class="p-4 py-2 rounded-full cursor-pointer bg-tab-active"
  >
    <p
      class="font-bold sm:text-sm text-xs"
    >
      TEST TITLE
    </p>
  </div>
</div>
`;
